main.cpp

```diff
@@ -1,179 +1,120 @@
         {
             board[i][j] = NULL;
         }
     }
 
-    // counters for the number of each Critter being placed.
+    //Counters for the number of each Critter being placed.
     int doodCount = 0;
     int antCount = 0;
 
-    //helper variable setting age to 8 for breed test
-    //int age = 8;
-
-    // Place Doodlebugs on the board
+    //Place Doodlebugs on the board
     placeDoodles(board, doodCount, numDoodbugs, size_x, size_y);
 
-    // Place Ants on the board
+    //Place Ants on the board
     placeAnts(board, antCount, numAnts, size_x, size_y);
 
-    //print starting board - this should be a function outside of main that just gets called
-    printBoard(board, size_y, size_x);
+    //Print starting board 
+    printBoard(board, size_x, size_y);
 
 
-    /*********************************************************************************
-    ** This area contains all the simulated movement/feeding, breeding, and starving
-    ** the Ants and the Doodlebugs will take part in.
-    *********************************************************************************/
-
-    int stepCounter = 0;
-
+/*********************************************************************************
+** This area contains all the simulated movement/feeding, breeding, and starving
+** the Ants and the Doodlebugs will take part in.
+*********************************************************************************/
     // loop through the simulation until the user decides to end the simulation
     while (timeSteps != 0)
     {
         
         /*************************************
         ** MOVE THE DOODLEBUGS
         *************************************/
+        moveCritters(board, size_x, size_y, 'X');
         
-        moveCritters(board, size_x, size_y, 'X', "doodlebug");        
         resetCritterMoveSuccess(board, size_x, size_y);
         
         /*************************************
          ** MOVE THE ANTS
          *************************************/
+        moveCritters(board, size_x, size_y, 'O');
         
-        moveCritters(board, size_x, size_y, 'O', "ant");
         resetCritterMoveSuccess(board, size_x, size_y);
 
-        /*************************************
-        ** BREED THE CRITTERS
-        *************************************/
-    
-        breedCritters(board, size_x, size_y, 'X', "doodlebug");
-        breedCritters(board, size_x, size_y, 'O', "ant");
-        
         /*************************************
         ** STARVE THE DOODLEBUGS
         *************************************/
-        starveCritters(board, size_x, size_y, 'X', "doodlebug");
+        starveCritters(board, size_x, size_y, 'X');
 
-        cout << "\nStep " << ++stepCounter << " Completed!\n";
-        cout << "\n" << "Printing board again" << "\n\n";
-
-        //print board again to see if breed worked
+        /*************************************
+        ** BREED THE CRITTERS
```

Comment:
is it supposed to be breeding or starving last?

Reply:
I think it should be starving before breeding only because if we have a Doodlebug who is say Age 9, but just hit the starvation in the move phase of the game, does it make sense to have it breed before it's removed from the board?